src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -137,25 +142,102 @@ public void addVar(Declaration.Variable varTree) {
         String layoutField = emitLayoutFieldDecl(varTree, javaName);
         appendBlankLine();
         String offsetField = emitOffsetFieldDecl(varTree, javaName);
-        if (Utils.isArray(varTree.type()) || Utils.isStructOrUnion(varTree.type())) {
+        Type type = varTree.type();
+        if (Utils.isArray(type) || Utils.isStructOrUnion(type)) {
             emitSegmentGetter(javaName, varTree, offsetField, layoutField);
             emitSegmentSetter(javaName, varTree, offsetField, layoutField);
-            int dims = Utils.dimensions(varTree.type()).size();
+            int dims = Utils.dimensions(type).size();
             if (dims > 0) {
                 emitDimensionsFieldDecl(varTree, javaName);
                 String arrayHandle = emitArrayElementHandle(javaName, varTree, layoutField, dims);
                 IndexList indexList = IndexList.of(dims);
                 emitFieldArrayGetter(javaName, varTree, arrayHandle, indexList);
                 emitFieldArraySetter(javaName, varTree, arrayHandle, indexList);
             }
-        } else if (Utils.isPointer(varTree.type()) || Utils.isPrimitive(varTree.type())) {
-            emitFieldGetter(javaName, varTree, layoutField, offsetField);
-            emitFieldSetter(javaName, varTree, layoutField, offsetField);
+        } else if (Utils.isPointer(type) || Utils.isPrimitive(type)) {
+            boolean isFuncPtr = functionalDispatch && Utils.isFunctionPointer(type);
+            if (isFuncPtr) {
+                emitFieldSetter(javaName, varTree, layoutField, offsetField);
```

Comment:
I'm not sure we even want the setter -- the fields in these structs are typically populated by a library